SYCL/ESIMD/api/functional/value.hpp

```diff
@@ -0,0 +1,185 @@
+      return half_from_bytes(0b01111100u, 0b00000000u);
+    } else {
+      return std::numeric_limits<DataT>::infinity();
+    }
+  }
+
+  static DataT lowest() {
+    if constexpr (std::is_same_v<DataT, sycl::half>) {
+      return -half_from_bytes(0b01111011u, 0b11111111u);
+    } else {
+      return std::numeric_limits<DataT>::lowest();
+    }
+  }
+
+  static DataT denorm_min() {
+    if constexpr (std::is_same_v<DataT, sycl::half>) {
+      return half_from_bytes(0b00000000u, 0b00000001u);
+    } else {
+      return std::numeric_limits<RefType>::denorm_min;
+    }
+  }
+
+  static DataT nan(unsigned char opcode = 42u) {
+    static_assert(
+        type_traits::is_floating_point_v<DataT>,
+        "nan can value can retrieved only for floating point variable type.");
+    if constexpr (std::is_same_v<DataT, double>) {
+      return sycl::nan(42UL);
+    } else if constexpr (std::is_same_v<DataT, float>) {
+      return sycl::nan(42U);
+    } else if constexpr (std::is_same_v<DataT, sycl::half>) {
+      return half_from_bytes(0b11111110u, 0b00000000u + opcode);
+    }
+  }
+
+  static DataT max() {
+    if constexpr (std::is_same_v<DataT, sycl::half>) {
+      return half_from_bytes(0b01111011u, 0b11111111u);
+    } else {
+      return std::numeric_limits<DataT>::max();
+    }
+  }
+};
+
+// Filling obtained std::vector with reference data according to current tested
+// data type and number of elements.
+template <typename DataT, int NumElems> std::vector<DataT> generate_ref_data() {
+  static_assert(
+      std::is_integral_v<DataT> || std::is_arithmetic_v<DataT> ||
+          type_traits::is_floating_point_v<DataT>,
+      "Invalid data type provided to the generate_ref_data function.");
+
+  static const DataT min = value<DataT>::lowest();
+  static const DataT min_half = min / 2;
+  static const DataT max = value<DataT>::max();
+  static const DataT max_half = max / 2;
+
+  std::vector<DataT> ref_data{};
+
+  if constexpr (std::is_signed_v<DataT>) {
+    ref_data.reserve((NumElems > 1) ? NumElems : 5);
+
+    ref_data.insert(ref_data.end(), {min, min_half, max, max_half, 0});
+    if constexpr (NumElems != 1) {
+      ref_data.insert(ref_data.end(), {static_cast<DataT>(min + 1),
+                                       static_cast<DataT>(max - 1), -1});
+      for (size_t i = ref_data.size(); i < NumElems; ++i) {
+        ref_data.push_back(i);
+      }
+    }
+  }
+
+  if constexpr (std::is_unsigned_v<DataT>) {
+    ref_data.reserve((NumElems > 1) ? NumElems : 3);
+
+    ref_data.insert(ref_data.end(), {max, max_half, 0});
+    if constexpr (NumElems != 1) {
+      ref_data.insert(ref_data.end(), {static_cast<DataT>(max - 1)});
+      for (size_t i = ref_data.size(); i < NumElems; ++i) {
+        ref_data.push_back(i);
+      }
+    }
+  }
+
+  if constexpr (type_traits::is_floating_point_v<DataT>) {
+    static const DataT nan{value<DataT>::nan()};
+    static const DataT inf = value<DataT>::inf();
```

Comment:
```suggestion
    static const DataT nan = value<DataT>::nan();
    static const DataT inf = value<DataT>::inf();
```
Please be consistent over the code if it possible. Avoid using the different ways for the same thing.

Reply:
this comment was applied in this commit: 4bd389b51699b9b2e1d891e8bcab1c536cd45a90